File: Selenium/Project/Activity9.java
package ibm.fst_m1_SeleniumProject;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;
import io.github.bonigarcia.wdm.WebDriverManager;

public class Activity9 {
    WebDriver driver;
    
    @BeforeTest(alwaysRun = true)
    public void beforeMethod() {
        // Set up the Firefox driver
        WebDriverManager.firefoxdriver().setup();
        //Create a new instance of the Firefox driver
        driver = new FirefoxDriver();   
        
        driver.manage().window().maximize();
        
        //Open the browser
        driver.get("https://alchemy.hguy.co/lms");
        

    }
    
    @Test (priority = 1)
   public void HeaderTest3() {
    	
    	WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    	driver.findElement(By.xpath("//*[@id=\"menu-item-1507\"]")).click();
    	
    	WebElement page_title = driver.findElement(By.cssSelector("h1.uagb-ifb-title"));
        Assert.assertEquals(page_title.getText(), "My Account");
        System.out.println("On Page: " + page_title.getText());
        
        driver.findElement(By.cssSelector("a.ld-login")).click();
        
        WebElement dynamicBox = driver.findElement(By.cssSelector("div.ld-login-modal-login"));
        wait.until(ExpectedConditions.visibilityOf(dynamicBox));
        System.out.println(dynamicBox.isDisplayed());
    }
    
    @Test (priority = 2)
    public void loginTest() {
        //Find the username and password fields
        WebElement username = driver.findElement(By.id("user_login"));
        WebElement password = driver.findElement(By.id("user_pass"));
        
        //Enter credentials
        username.sendKeys("root");
        password.sendKeys("pa$$w0rd");
        
        //Click login
        driver.findElement(By.xpath("//input[@id='wp-submit']")).click();
        
       driver.findElement(By.cssSelector("img.avatar-96"));
       System.out.println("Logged In Successful");
        
       }    

    @Test (priority = 3)
    public void CourseCompletion() {
        
    	//Click on All Courses link
    	driver.findElement(By.xpath("//*[@id=\"menu-item-1508\"]")).click();
        
    	driver.findElement(By.cssSelector("a.btn.btn-primary")).click();
        
    	System.out.println("Mark Complete Button is not found on the page");
        
       } 
    
	@AfterTest(alwaysRun = true)
    public void afterMethod() {
        //Close the browser
       driver.close();
    }

}

File: Selenium/Activities/activity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys

with webdriver.Firefox() as driver:
    driver.get("https://v1.training-support.net/selenium/ajax")
    print("Title is: ",driver.title)
    actions = ActionChains(driver)
    ChangeContent=driver.find_element(By.CSS_SELECTOR,"button.violet")
    wait = WebDriverWait(driver, 10)
    ChangeContent.click()
    wait.until(expected_conditions.element_to_be_clickable(driver.find_element(By.XPATH,"//div[@id='ajax-content']/h1")))
    h1=driver.find_element(By.XPATH,"//div[@id='ajax-content']/h1").text
    h3=driver.find_element(By.XPATH,"//div[@id='ajax-content']/h3").text
    print("H1 Text is ",h1)
    print("H3 Text before change ",h1)
    wait.until(expected_conditions.element_to_be_clickable(driver.find_element(By.XPATH,"//div[@id='ajax-content']/h3")))
    h3_new=driver.find_element(By.XPATH,"//div[@id='ajax-content']/h3").text
    print("H3 Text after change ",h3_new)